### PR TITLE
Implement column resizing

### PR DESCRIPTION
**Pull request by FadhlanR**
https://github.com/cardstack/boxel/assets/12637010/3a578a71-addf-4fa0-b588-f9edde899ee5

- [x] Resizing by dragging the resizer button.
- [x] Open/close the left/right panel by double-clicking the resizer button.
- [x] Persist this via local storage (applies to all of the code mode).

*) The resizer button icon needs to be updated.

### CODE REVIEW

**Review by tintinthong — APPROVED** [review]
ive tested and it works great

**Review by jurgenwerk — COMMENTED** [review]
I tested this and it works nice, great job!

There is one thing that is bothering me-I think the whole logic for this layout feature should be moved out of the code-mode component otherwise we risk this file getting huge and having too many responsibilities. I believe ideally code mode component should stick to managing data flows while layout logic is dealt with in another place.

Perhaps you can consider extracting this resizing logic somehow. I'm wondering if a combination of a modifier and state service would work. Maybe you could implement a modifier which would enhance the columns with the resizing feature when attached to the code mode component. One other option could be a new `ResizableCodeModeColumns` component with named blocks for columns

**Comment by FadhlanR** [comment]
> I tested this and it works nice, great job!
> 
> There is one thing that is bothering me-I think the whole logic for this layout feature should be moved out of the code-mode component otherwise we risk this file getting huge and having too many responsibilities. I believe ideally code mode component should stick to managing data flows while layout logic is dealt with in another place.
> 
> Perhaps you can consider extracting this resizing logic somehow. I'm wondering if a combination of a modifier and state service would work. Maybe you could implement a modifier which would enhance the columns with the resizing feature when attached to the code mode component. One other option could be a new `ResizableCodeModeColumns` component with named blocks for columns

I've decided to create a new component in Boxel UI called 'resizable-panel'.


https://github.com/cardstack/boxel/assets/12637010/dd05f78e-332b-4e7b-8173-a4d840336802

**Review by habdelra — APPROVED** [review]
great job!

File: packages/host/app/components/operator-mode/code-mode.gts [hunk 1]
```diff
@@ -155,8 +186,16 @@ export default class CodeMode extends Component<Signature> {
       class='card-url-bar'
     />
     <div class='code-mode' data-test-code-mode>
-      <div class='columns'>
-        <div class='column'>
```
Comment (burieberry) [inline packages/host/app/components/operator-mode/code-mode.gts:230]: Why did you delete the containers inside the first column? In the latest design I see two containers in the first column: https://app.zeplin.io/project/64c273904a22522215257505/screen/64ecae6356d7a924c50cbddd. It should be possible to change the width of the column still.

File: packages/host/app/components/operator-mode/code-mode.gts [hunk 2]
```diff
@@ -16,25 +16,47 @@ import CardService from '@cardstack/host/services/card-service';
...
     super(args, owner);
     this.fetchCodeModeRealmInfo.perform();
+
+    this.panelWidths = localStorage.getItem(CodeModePanelWidths)
+      ? // @ts-ignore Type 'null' is not assignable to type 'string'
+        JSON.parse(localStorage.getItem(CodeModePanelWidths))
+      : {
+          rightPanel: '25%',
+          codeEditorPanel: '50%',
+          leftPanel: '25%',
```
Comment (burieberry) [inline packages/host/app/components/operator-mode/code-mode.gts]: I'd say that the default ratios look more like this: if left panel is x, middle panel is 3x, and right panel is 2x

**Review by burieberry — COMMENTED** [review]
The resizing handle works well, but there seems to be something off with the column widths. I get something like this:

columns

File: packages/host/app/components/operator-mode/code-mode.gts (hunk 1 above)
Comment (FadhlanR) [reply packages/host/app/components/operator-mode/code-mode.gts:230]: Nice catch! I made a mistake here.

File: packages/host/app/components/operator-mode/code-mode.gts (hunk 2 above)
Comment (FadhlanR) [reply packages/host/app/components/operator-mode/code-mode.gts]: Updated here [703363c](https://github.com/cardstack/boxel/pull/600/commits/703363c6ec76cd6b7c2eb423e195587ec1e980b3)

File: packages/host/app/components/operator-mode/code-mode.gts (hunk 1 above)
Comment (FadhlanR) [reply packages/host/app/components/operator-mode/code-mode.gts:230]: Updated here [703363c](https://github.com/cardstack/boxel/pull/600/commits/703363c6ec76cd6b7c2eb423e195587ec1e980b3)

**Comment by FadhlanR** [comment]
> The resizing handle works well, but there seems to be something off with the column widths. I get something like this:
> 
> columns

Fixed!

https://github.com/cardstack/boxel/assets/12637010/2a7d5a4e-bcaf-4bcc-950e-62925afb1012

**Comment by FadhlanR** [comment]
After updating the resize handler button:


https://github.com/cardstack/boxel/assets/12637010/8d36d616-e737-48fd-bec2-30ec06e00e62

**Review by burieberry — APPROVED** [review]
Great
